gradle-mvn-push.gradle

```diff
@@ -1,228 +1,155 @@
+					from sourceSets.testFixtures.allSource
+				}
+
+				publishing {
+					publications {
+						wala(MavenPublication) {
+							artifact testFixturesJavadocJar
+							artifact testFixturesSourcesJar
+						}
+					}
+				}
+			}
+
+			pom {
+				name = POM_NAME
+				description = 'T. J. Watson Libraries for Analysis'
+				inceptionYear = '2006'
+				url = 'https://github.com/wala/WALA'
+
+				ciManagement {
+					system = 'Travis CI'
+					url = 'https://travis-ci.org/wala/WALA'
+				}
+
+				developers {
+					[
+							// Current WALA maintainers, alphabetical by ID
+							juliandolby: 'Julian Dolby',
+							liblit     : 'Ben Liblit',
+							msridhar   : 'Manu Sridharan',
+							sjfink     : 'Stephen Fink',
+					].each { entry ->
+						developer {
+							id = entry.key
+							name = entry.value
+							url = "https://github.com/$entry.key"
+						}
+					}
+				}
+
+				issueManagement {
+					system = 'GitHub'
+					url = "${pom.url.get()}/issues"
+				}
+
+				licenses {
+					license {
+						name = 'Eclipse Public License v2.0'
+						url = "${pom.url.get()}/blob/master/LICENSE"
+					}
+				}
+
+				mailingLists {
+					[
+							'commits',
+							'wala',
+					].each { topic ->
+						mailingList {
+							name = "wala-$topic"
+							archive = "https://sourceforge.net/p/wala/mailman/wala-$topic"
+							subscribe = "https://sourceforge.net/projects/wala/lists/wala-$topic"
+							unsubscribe = "https://sourceforge.net/projects/wala/lists/wala-$topic/unsubscribe"
+							post = "wala-$topic@lists.sourceforge.net"
+						}
+					}
+				}
+
+				scm {
+					url = pom.url
+					connection = 'scm:git:git://github.com/wala/WALA.git'
+					developerConnection = 'scm:git:ssh://git@github.com/wala/WALA.git'
+				}
+			}
+		}
+	}
+
+	repositories {
+		maven {
+			url = (isSnapshot
+					? properties.getOrDefault('SNAPSHOT_REPOSITORY_URL', 'https://oss.sonatype.org/content/repositories/snapshots/')
+					: properties.getOrDefault('RELEASE_REPOSITORY_URL', 'https://oss.sonatype.org/service/local/staging/deploy/maven2/')
+			)
+			credentials {
+				username = properties.get('SONATYPE_NEXUS_USERNAME')
+				password = properties.get('SONATYPE_NEXUS_PASSWORD')
+			}
+		}
+
+		maven {
+			name = 'projectBuild'
+			url = "file://$rootProject.buildDir/maven-repository"
+		}
+	}
 }
 
-def getReleaseRepositoryUrl() {
-  return hasProperty('RELEASE_REPOSITORY_URL') ? RELEASE_REPOSITORY_URL
-          : "https://oss.sonatype.org/service/local/staging/deploy/maven2/"
+signing {
+	// Use external gpg cmd.  This makes it easy to use gpg-agent,
+	// to avoid being prompted for a password once per artifact
+	useGpgCmd()
+	sign publishing.publications.wala
+	required = true
 }
 
-def getSnapshotRepositoryUrl() {
-  return hasProperty('SNAPSHOT_REPOSITORY_URL') ? SNAPSHOT_REPOSITORY_URL
-          : "https://oss.sonatype.org/content/repositories/snapshots/"
+// Only sign the archives if we are uploading a snapshot or release.
```

Comment:
We don’t want to sign snapshots. I think just the comment is incorrect here

Reply:
Good catch. Comment corrected in beed95b863238b967100014d46e517f8eb2cf1fa, then reworked again in de3911d0060b283195a7fab91b7f59d1d43ecdde along with other changes.